source/simulator/assembly.cc

```diff
@@ -936,6 +942,14 @@ namespace aspect
     for (unsigned int q=0; q<n_q_points; ++q)
       for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
         material_model_inputs.composition[q][c] = composition_values[c][q];
+
+    typename Triangulation<dim>::active_cell_iterator cell = input_finite_element_values.get_cell();
+    typename DoFHandler<dim>::active_cell_iterator dof_cell(&triangulation,
+                                                            cell->level(),
+                                                            cell->index(),
+                                                            &dof_handler);
+
+    material_model_inputs.cell = &dof_cell;
   }
```

Comment:
This won't work. You take a reference to a temporary variable that goes out of scope at the curly brace right here. You need to declare `dof_cell` at a location where its lifetime is the same as that of the `material_model_inputs` object.